lib/vmselectapi/server.go

```diff
@@ -919,6 +923,9 @@ func (s *Server) processTenants(ctx *vmselectRequestCtx) error {
 
 	// Send tenants to vmselect
 	for _, tenant := range tenants {
+		if len(tenant) == 0 {
+			return fmt.Errorf("BUG: unexpected empty tenant name")
```

Comment:
I think it is better to just skip an empty tenant and log this issue instead of returning the error, since this error makes the tenant API unusable.